Add `wcadmin_pfw_domain_verify_success` event

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Fix `pfw_domain_verify_failure` tests, 
   Mock ESM default export, make the mock sync.
- Fire `wcadmin_pfw_domain_verify_success` event, 
   implements part of #232


### Screenshots:

<!--- Optional --->

![verify](https://user-images.githubusercontent.com/17435/147793787-cc9a262e-b1d9-4c9e-a64e-fc7201b8c4d2.gif)


### Detailed test instructions:

0. Start ngrok, or other tunnel to publish your local site.
1. Enable track events logging, with `localStorage.setItem( 'debug', 'wc-admin:*' );`
2. Disconnect your account if needed `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fconnection`
1. Start the onboarding `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding`
2. Follow the wizard, to step 2
3. Click "Start verification"
4. Once successfully verified, you should see `wcadmin_pfw_domain_verify_success` fired.



<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

tomalec committed:
Fix `pfw_domain_verify_failure` tests,

Mock ESM default export, make the mock sync.

tomalec committed:
Fire `wcadmin_pfw_domain_verify_success` event,

implements part of https://github.com/woocommerce/pinterest-for-woocommerce/issues/232

tomalec committed:
Re-generate TRACKING.md docs

tomalec commented:
After this PR is merge the feature branch PR should be ready for the review https://github.com/woocommerce/pinterest-for-woocommerce/pull/296

puntope commented on assets/source/setup-guide/app/steps/ClaimWebsite.test.js:
```diff
@@ -48,4 +57,25 @@ describe( 'Claim Website Record Events', () => {
+
+		fireEvent.click( getByText( 'Start verification' ) );
+
+		// Wait for async click handler and apiFetch resolution.
+		const delay = sleep( 1 );
+		jest.runAllTimers();
+		await delay;
+
+		expect( recordEvent ).toHaveBeenCalledWith(
+			'pfw_domain_verify_success'
+		);
```
💅 

WDYT about...

```
await waitFor( () =>
			expect( recordEvent ).toHaveBeenCalledWith(
				'pfw_domain_verify_success'
			)
		);
```

more info [here](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor)

tomalec replied:
I like it :)
`setTimout(1)` was testing that it's fired immediately after, but `waitFor` should be less fragile` is more readable, and shorter.

Done in https://github.com/woocommerce/pinterest-for-woocommerce/pull/314/commits/de3d9e7646373829b3b942adef0626afc8b06098

We could think of something more like
```js
await waitForNetworkIdle(); // await waifForApiFetchMock()
expect( //...
```
to make sure we fire it synchronously after  API fetch response. But let's not overdo it.

puntope commented on assets/source/setup-guide/app/steps/ClaimWebsite.test.js:
```diff
@@ -32,9 +33,17 @@ import { fireEvent, render } from '@testing-library/react';
  */
 import ClaimWebsite from './ClaimWebsite';
```
💅  

Could we remove as well

`jest.mock( '@woocommerce/tracks'`

I guess I forgot myself to remove it when we implemented this in `test/dependencies` as discussed.

puntope reviewed (commented):
✅  LGTM

✅  I tested the domain verification and worked for me :)
✅  Visual code verification looks good 

💅  Some opinionated comments regarding the tests

tomalec committed:
Replace `setTimout` with `waitFor`

in `pfw_domain_verify_success` test.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/314#discussion_r776989581

tomalec committed:
Remove redundant `@woocommerce/tracks` mock.

It's already mocked by `jest.config`'s `moduleNameMapper`.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/314#discussion_r776990038

tomalec commented:
Sorry for the force pushes, I struggled with accidentally staged `.pot` file

tomalec commented:
Thanks for the review @puntope I pushed both of your suggestions. Would you mind another look?
